roachtest: sqlalchemy failed

[(roachtest).sqlalchemy failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1801614&tab=buildLog) on [master@5570c01402796edb7cd06eb8ce7f615371f22d42](https://github.com/cockroachdb/cockroach/commits/5570c01402796edb7cd06eb8ce7f615371f22d42):

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/20200311-1801614/sqlalchemy/run_1
	orm_helpers.go:214,orm_helpers.go:144,sqlalchemy.go:220,sqlalchemy.go:231,test_runner.go:747: 
		Tests run on Cockroach v20.1.0-beta.2-737-g5570c01
		Tests run against sqlalchemy rel_1_3_14
		209 Total Tests Run
		134 tests passed
		75 tests failed
		107 tests skipped
		1 test ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: test/dialect/test_suite.py::ExceptionTest_cockroachdb+psycopg2_9_5_0::test_integrity_error (unexpected)
		For a full summary look at the sqlalchemy artifacts 
		
		An updated blacklist (sqlAlchemyBlacklist20_1) is available in the artifacts' sqlalchemy log

```

<details><summary>More</summary><p>


Artifacts: [/sqlalchemy](https://teamcity.cockroachdb.com/viewLog.html?buildId=1801614&tab=artifacts#/sqlalchemy)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Asqlalchemy.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).sqlalchemy failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1804460&tab=buildLog) on [master@793a9200c16693aff32aa6a4dd9d8bbcbddb30aa](https://github.com/cockroachdb/cockroach/commits/793a9200c16693aff32aa6a4dd9d8bbcbddb30aa):

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/20200312-1804460/sqlalchemy/run_1
	orm_helpers.go:214,orm_helpers.go:144,sqlalchemy.go:220,sqlalchemy.go:231,test_runner.go:747: 
		Tests run on Cockroach v20.1.0-beta.2-770-g793a920
		Tests run against sqlalchemy rel_1_3_15
		209 Total Tests Run
		134 tests passed
		75 tests failed
		107 tests skipped
		1 test ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: test/dialect/test_suite.py::ExceptionTest_cockroachdb+psycopg2_9_5_0::test_integrity_error (unexpected)
		For a full summary look at the sqlalchemy artifacts 
		
		An updated blacklist (sqlAlchemyBlacklist20_1) is available in the artifacts' sqlalchemy log

```

<details><summary>More</summary><p>


Artifacts: [/sqlalchemy](https://teamcity.cockroachdb.com/viewLog.html?buildId=1804460&tab=artifacts#/sqlalchemy)
Related:
- #46025 roachtest: sqlalchemy failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.2) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Asqlalchemy.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/cmd/roachtest/sqlalchemy.go
+++ b/pkg/cmd/roachtest/sqlalchemy.go
@@ -95,7 +95,8 @@ func registerSQLAlchemy(r *testRegistry) {
 			c,
 			node,
 			"install pytest",
-			`sudo pip3 install --upgrade --force-reinstall setuptools pytest pytest-xdist psycopg2`,
+			// Unpin pytest once sqlalchemy rel_1_3_16 is released (https://github.com/sqlalchemy/sqlalchemy/issues/5201)
+			`sudo pip3 install --upgrade --force-reinstall setuptools pytest==5.3.5 pytest-xdist psycopg2`,
 		); err != nil {
 			t.Fatal(err)
 		}

--- a/pkg/cmd/roachtest/sqlalchemy_blacklist.go
+++ b/pkg/cmd/roachtest/sqlalchemy_blacklist.go
@@ -160,5 +160,6 @@ var sqlAlchemyBlacklist = blacklist{
 var sqlAlchemyIgnoreList20_1 = sqlAlchemyIgnoreList
 
 var sqlAlchemyIgnoreList = blacklist{
-	"test/dialect/test_suite.py::TableDDLTest_cockroachdb+psycopg2_9_5_0::test_create_table": "flaky",
+	"test/dialect/test_suite.py::ExceptionTest_cockroachdb+psycopg2_9_5_0::test_integrity_error": "passes, but can't parse result",
+	"test/dialect/test_suite.py::TableDDLTest_cockroachdb+psycopg2_9_5_0::test_create_table":     "flaky",
 }

